fdbserver/workloads/AtomicOps.actor.cpp

```diff
@@ -172,29 +182,138 @@ struct AtomicOpsWorkload : TestWorkload {
 			loop {
 				try {
 					int group = deterministicRandom()->randomInt(0,100);
-					uint64_t intValue = deterministicRandom()->randomInt( 0, 10000000 );
+					state uint64_t intValue = deterministicRandom()->randomInt(0, 10000000);
 					Key val = StringRef((const uint8_t*) &intValue, sizeof(intValue));
-					tr.set(self->logKey(group), val);
+					std::pair<Key, Key> logDebugKey = self->logDebugKey(group);
 					int nodeIndex = deterministicRandom()->randomInt(0, self->nodeCount / 100);
-					tr.atomicOp(StringRef(format("ops%08x%08x", group, nodeIndex)), val, self->opType);
-					// TraceEvent(SevDebug, "AtomicOpWorker")
-					//     .detail("LogKey", self->logKey(group))
-					//     .detail("Value", val)
-					//     .detail("ValueInt", intValue);
-					// TraceEvent(SevDebug, "AtomicOpWorker")
-					//     .detail("OpKey", format("ops%08x%08x", group, nodeIndex))
-					//     .detail("Value", val)
-					//     .detail("ValueInt", intValue)
-					//     .detail("AtomicOp", self->opType);
+					Key opsKey(format("ops%08x%08x", group, nodeIndex));
+					tr.set(logDebugKey.first, val); // set log key
+					tr.set(logDebugKey.second, opsKey); // set debug key; one opsKey can have multiple logs key
+					tr.atomicOp(opsKey, val, self->opType);
 					wait( tr.commit() );
+					if (self->opType == MutationRef::AddValue) {
+						self->lbsum += intValue;
+						self->ubsum += intValue;
+					}
 					break;
 				} catch( Error &e ) {
 					wait( tr.onError(e) );
+					if (self->opType == MutationRef::AddValue) {
+						self->ubsum += intValue;
```

Comment:
Yes, that is part of the reason.
Another reason is to tell if the value calculated in the `_check` is larger or smaller than the  value calculated in the workload.

Reply:
The scenario is:
The test fails and says `ops` sum value is inconsistent with the `logs` value. Is it because `ops` were applied multiple times or because some `logs` values are lost?
The `ubsum` and `lbsum` is to provide more information in the above scenario.

Reply:
My question is more what specific conditions cause the upper bound increase here but not the lower bound increase. The only case I thought of was the one I listed above, but if there are others my suggestion may be less applicable.

Reply:
I think this question is still open.

Reply:
I see your point now. Yes, that's the only condition when `ubsum` is not equal to `lbsum`. I can change this and add the trace event as you suggested.